app/src/main/java/com/github/se/eventradar/viewmodel/HostedEventsViewModel.kt

```diff
@@ -23,40 +25,51 @@ constructor(
   private val _uiState = MutableStateFlow(HostedEventsUiState())
   val uiState: StateFlow<HostedEventsUiState> = _uiState
 
-  fun getHostedEvents(uid: String) {
-    viewModelScope.launch {
-      when (val userResponse = userRepository.getUser(uid)) {
-        is Resource.Success -> {
-          val user = userResponse.data!!
-          val eventsHostList = user.eventsHostSet.toList()
-          if (eventsHostList.isNotEmpty()) {
-            when (val events = eventRepository.getEventsByIds(eventsHostList)) {
-              is Resource.Success -> {
-                _uiState.value =
-                    _uiState.value.copy(
-                        eventList =
-                            EventList(
-                                events.data, events.data, _uiState.value.eventList.selectedEvent))
-              }
-              is Resource.Failure -> {
-                Log.d("HostedEventsViewModel", "Error getting hosted events for $uid")
-                _uiState.value =
-                    _uiState.value.copy(eventList = EventList(emptyList(), emptyList(), null))
+  fun getHostedEvents(uid: String? = Firebase.auth.currentUser?.uid) {
+    if (uid == null) {
+      Log.d("HostedEventsViewModel", "User not logged in")
+    } else {
+      viewModelScope.launch {
+        when (val userResponse = userRepository.getUser(uid)) {
+          is Resource.Success -> {
+            val user = userResponse.data!!
+            val eventsHostSet = user.eventsHostSet
+            if (eventsHostSet.isNotEmpty()) {
+              when (val events = eventRepository.getEventsByIds(eventsHostSet.toList())) {
+                is Resource.Success -> {
+                  _uiState.value =
+                      _uiState.value.copy(
+                          eventList =
+                              EventList(
+                                  events.data, events.data, _uiState.value.eventList.selectedEvent))
+                }
+                is Resource.Failure -> {
+                  Log.d("HostedEventsViewModel", "Error getting hosted events for $uid")
+                  _uiState.value =
+                      _uiState.value.copy(eventList = EventList(emptyList(), emptyList(), null))
+                }
               }
+            } else {
+              _uiState.value =
+                  _uiState.value.copy(eventList = EventList(emptyList(), emptyList(), null))
             }
-          } else {
+          }
+          is Resource.Failure -> {
+            Log.d("HostedEventsViewModel", "Error fetching user document")
             _uiState.value =
                 _uiState.value.copy(eventList = EventList(emptyList(), emptyList(), null))
           }
         }
-        is Resource.Failure -> {
-          Log.d("HostedEventsViewModel", "Error fetching user document")
-          _uiState.value =
-              _uiState.value.copy(eventList = EventList(emptyList(), emptyList(), null))
-        }
       }
     }
   }
+
+  fun onViewListStatusChanged(
```

Comment:
nit: this doesn't need to take in the boolean. Whenever it's called, we know the new value will be the opposite of what it is now.